crypto/crypto.go

```diff
@@ -34,11 +35,11 @@ func Sign(key SigningKey, message []byte) []byte {
 	return ed25519.Sign(ed25519.PrivateKey(key), message)
 }
 
-func Verify(key SigningKey, message, sig []byte) bool {
-	pk, ok := ed25519.PrivateKey(key).Public().(ed25519.PublicKey)
-	if !ok {
-		return false
-	}
+func SigningKeyToPublic(key SigningKey) []byte {
+	return ed25519.PrivateKey(key).Public().(ed25519.PublicKey)
+}
+
+func Verify(pk, message, sig []byte) bool {
 	return ed25519.Verify(pk, message, sig)
 }
```

Comment:
Since we already have a type, what if we did something like this instead?

```
--- a/crypto/crypto.go
+++ b/crypto/crypto.go
@@ -13,8 +13,6 @@ const (
        PrivateKeySize = 64
 )
 
-type SigningKey ed25519.PrivateKey
-
 func Digest(ms ...[]byte) []byte {
        h := sha3.NewShake128()
        for _, m := range ms {
@@ -25,21 +23,23 @@ func Digest(ms ...[]byte) []byte {
        return ret
 }
 
-func GenerateKey() (SigningKey, error) {
+type SigningKey ed25519.PrivateKey
+
+func NewSigningKey() (SigningKey, error) {
        _, sk, err := ed25519.GenerateKey(rand.Reader)
        return SigningKey(sk), err
 }
 
-func Sign(key SigningKey, message []byte) []byte {
+func (key SigningKey) Sign(message []byte) []byte {
        return ed25519.Sign(ed25519.PrivateKey(key), message)
 }
 
-func Verify(key SigningKey, message, sig []byte) bool {
-       pk, ok := ed25519.PrivateKey(key).Public().(ed25519.PublicKey)
-       if !ok {
-               return false
-       }
-       return ed25519.Verify(pk, message, sig)
+func (key SigningKey) Public() []byte {
+       return ed25519.PrivateKey(key).Public().(ed25519.PublicKey)
+}
+
+func (key SigningKey) Verify(message, sig []byte) bool {
+       return ed25519.Verify(key, message, sig)
 }
```